### PR TITLE
Stop repeatedly detecting changed properties on Deleted entities

### LINKED ISSUE #21896: "PropertyChangeDetected" Event Logged More Than Once
I don't know if this a bug or by-design.

``` C#
        static void Main(string[] args)
        {
            var context = new Context();

            var user = new User();

            context.Add(user);

            context.SaveChanges();

            user.Nickname = "foo";

            context.ChangeTracker.DetectChanges();

            context.Remove(user);

            Console.Clear();

            for (int i = 0; i < 100; i++)
                context.ChangeTracker.DetectChanges(); 
        }
```
 
The event is logged several times: 
```
dbug: Microsoft.EntityFrameworkCore.ChangeTracking[i]
      Unchanged 'User.Nickname' detected as changed from '(null)' to 'foo' and will be marked as modified for entity with key '{Id: 5cb136b6-2e2e-433e-ad22-b6813731769a}'.
```

### Further technical details
EF Core version: 3.1.6
Database provider: Any
Target framework: NET Core 3.0

[Solution.zip](https://github.com/dotnet/efcore/files/5011085/Solution.zip)

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #21896

### CODE REVIEW

**Comment by ajcvickers** [comment]
@Pilchie RC1

File: src/EFCore/ChangeTracking/Internal/ChangeDetector.cs [hunk 1]
```diff
@@ -83,7 +84,14 @@ public virtual void PropertyChanged(InternalEntityEntry entry, IPropertyBase pro
...
             if (propertyBase is IProperty property)
             {
-                entry.SetPropertyModified(property, setModified);
+                if (entry.EntityState != EntityState.Deleted)
+                {
+                    entry.SetPropertyModified(property, setModified);
+                }
+                else
+                {
+                    ThrowIfKeyChanged(entry, property);
```
Comment (AndriySvyryd) [inline src/EFCore/ChangeTracking/Internal/ChangeDetector.cs:93]: Consider https://github.com/dotnet/efcore/issues/19135

Comment (ajcvickers) [reply src/EFCore/ChangeTracking/Internal/ChangeDetector.cs:93]: Added a note there. I don't think it's something we want to pull into 5.0.

**Comment by Pilchie** [comment]
Approved for RC1.

**Pull request merged**